src/requests/parameter_wrappers.rs

```diff
@@ -1,226 +1,177 @@
+        CqlType::Set => CqlValue::Set(cql_value_vec_from_array(
+            typ.support_type_1
+                .as_ref()
+                .expect("Expected support type for list"),
+            &get_element!(Array),
+        )?),
+        CqlType::UserDefinedType => todo!(),
+        CqlType::SmallInt => CqlValue::SmallInt(
+            get_element!(i32)
+                .try_into()
+                .map_err(|_| js_error("Value must fit in i16 type to be small int"))?,
+        ),
+        CqlType::TinyInt => CqlValue::TinyInt(
+            get_element!(i32)
+                .try_into()
+                .map_err(|_| js_error("Value must fit in ii8 type to be small int"))?,
+        ),
+        CqlType::Time => CqlValue::Time(get_element!(&LocalTimeWrapper).get_cql_time()),
+        CqlType::Timeuuid => CqlValue::Timeuuid(CqlTimeuuid::from_bytes(
+            get_element!(&UuidWrapper).get_cql_uuid().into_bytes(),
+        )),
+        CqlType::Tuple => todo!(),
+        CqlType::Uuid => CqlValue::Uuid(get_element!(&UuidWrapper).get_cql_uuid()),
+        CqlType::Varint => todo!(),
+        CqlType::Custom => unreachable!("Should not receive Custom type here"),
+        _ => todo!(),
+    };
+    Ok(value)
 }
 
-#[napi]
-impl MaybeUnsetQueryParameterWrapper {
-    #[napi]
-    pub fn from_non_null_non_unset_value(
-        val: &QueryParameterWrapper,
-    ) -> MaybeUnsetQueryParameterWrapper {
-        MaybeUnsetQueryParameterWrapper {
-            parameter: MaybeUnset::Set(val.parameter.clone()),
-        }
+impl FromNapiValue for ParameterWrapper {
+    /// # Safety
+    ///
+    /// Valid pointer to napi env must be provided
+    unsafe fn from_napi_value(
+        env: napi::sys::napi_env,
+        napi_val: napi::sys::napi_value,
+    ) -> napi::Result<Self> {
+        let parsing_error = || {
+            napi::Error::new(
+                Status::InvalidArg,
+                "Unexpected data when parsing parameters row".to_owned(),
+            )
+        };
+
+        // Caller of this function ensures a valid pointer to napi env is provided
+        let elem = unsafe { Array::from_napi_value(env, napi_val)? };
+        let val = if elem.len() == 0 {
+            None
+        } else if elem.len() == 2 {
+            let typ = elem.get::<&ComplexType>(0)?.ok_or(parsing_error())?;
+            if typ.is_unset {
+                Some(MaybeUnset::Unset)
+            } else {
+                Some(MaybeUnset::Set(cql_value_from_napi_value(typ, &elem, 1)?))
+            }
+        } else {
+            return Err(parsing_error());
+        };
+
+        Ok(ParameterWrapper { row: val })
     }
+}
 
-    #[napi]
-    pub fn unset() -> MaybeUnsetQueryParameterWrapper {
-        MaybeUnsetQueryParameterWrapper {
-            parameter: MaybeUnset::Unset,
-        }
-    }
+pub(crate) fn convert_parameters_row_wrapper(
+    val: Vec<ParameterWrapper>,
+) -> Vec<Option<MaybeUnset<CqlValue>>> {
+    val.into_iter().map(|e| e.row).collect()
 }
```

Comment:
:wrench: I don't understand the name of this function. Please rename it and introduce a comment, when it is used and what for.